packages/flutter/lib/src/foundation/platform.dart

```diff
@@ -52,3 +57,21 @@ TargetPlatform get defaultTargetPlatform {
   }
   return result;
 }
+/// Override the [defaultTargetPlatform].
+///
+/// Setting this to null returns the [defaultTargetPlatform] to its original
+/// value (based on the actual current platform).
+///
+/// This setter is only available in debug mode. To change the target platform
+/// in release builds, use the platform override APIs (such as
+/// [ThemeData.platform] in the material library).
+set defaultTargetPlatform(TargetPlatform value) {
```

Comment:
I'd make this a real function (e.g., `debugSetDefaultTargetPlatformOverride`) and throw a `NotImplemented` or some other normal sort of exception in release mode.